tools/sql_extraction/src/main/kotlin/DataFlowEngine.kt

```diff
@@ -3,12 +3,16 @@ package com.google.cloud.sqlecosystem.sqlextraction
 import com.google.cloud.sqlecosystem.sqlextraction.output.Location
 import com.google.cloud.sqlecosystem.sqlextraction.output.QueryFragment
 import com.google.cloud.sqlecosystem.sqlextraction.output.QueryUsages
+import mu.KotlinLogging
+
+private val LOGGER = KotlinLogging.logger { }
 
 /**
  * Backend engine for running data-flow analysis
+ *
  * @see FrontEnd
  */
-class DataFlowEngine {
+class DataFlowEngine(private val environment: Environment = Environment()) {
     private val queryUsages: HashMap<QueryFragment, HashSet<Location>> = HashMap()
```

Comment:
nit: in java we usually just use Map<QueryFragment, HashSet<Location>> = HashMap() if it is the same in kotlin..

Reply:
Map is immutable in Kotlin, but I can change it to MutableMap